### PR TITLE
fixed is_integer of Mul by not using fraction but collecting numerators and denominators directly

### LINKED ISSUE #20161: Inconsistent behavior for sympify/simplify with ceiling
In sympy v1.5.1:
```python
In [16]: sympy.sympify('4*ceiling(x/4 - 3/4)', evaluate=False).simplify()
Out[16]: 4*ceiling(x/4 - 3/4)

In [17]: sympy.sympify('4*ceiling(x/4 - 3/4)', evaluate=True).simplify()
Out[17]: 4*ceiling(x/4 - 3/4)
```

In sympy v.1.6.2:
```python
In [16]: sympy.sympify('4*ceiling(x/4 - 3/4)', evaluate=False).simplify()
Out[16]: 4*ceiling(x/4) - 3

In [17]: sympy.sympify('4*ceiling(x/4 - 3/4)', evaluate=True).simplify()
Out [17]: 4*ceiling(x/4 - 3/4)
```

Is there a way to ensure that the behavior is consistent, even though evaluate is equal to `False` when parsing?

### PR DESCRIPTION
**Pull request by coproc**
#### References to other Issues or PRs
Fixes #20161

#### Brief description of what is fixed or changed
In Mul._eval_is_integer instead of using the function fraction the args are scanned for contributions to the numerator and denominator.

#### Release Notes

* core
  * Fixed a few broken cases of expr.is_integer

### CODE REVIEW

File: sympy/core/mul.py [hunk 1]
```diff
@@ -1262,27 +1262,47 @@ def _eval_is_zero(self):
...
+                n, d = a.as_numer_denom()
+                numerators.append(n)
+                denominators.append(d)
+            elif a.is_Pow:
+                b, e = a.as_base_exp()
+                if not b.is_integer or not e.is_integer: return
+                if e.is_negative:
+                    denominators.append(b)
+                elif e.is_positive:
+                    denominators.append(b)
```
Comment (oscarbenjamin) [inline sympy/core/mul.py:1292]: I think this needs an `else: return`. If we don't know whether e is positive or negative then we don't know if b is in the numerator or denominator so the subsequent logic is not valid.

Comment (coproc) [reply sympy/core/mul.py]: You are right. I was implicitely assuming e.is_zero. Also the case e.is_positive is not needed - a.is_integer would already be triggered. (and the command inside was wrong: 'numerators.append(b)` would be needed)

**Comment by tbennun** [comment]
@oscarbenjamin @coproc sorry to bother again, but is there anything blocking this?

Thanks for the fix!

**Comment by oscarbenjamin** [comment]
Nothing is blocking this but a PR should also be opened against the 1.7 branch to make sure that the fix is in the 1.7 release.

**Comment by tbennun** [comment]
Anything I can do to help? There's also #20315 that's open against the 1.7 branch but it's probably superseded by this one.

**Comment by oscarbenjamin** [comment]
You could cherry pick the commits here for the 1.7 branch. I'll close #20315.

**Comment by tbennun** [comment]
@oscarbenjamin Done in #20450

**Comment by oscarbenjamin** [comment]
Thanks. This looks good!

**Pull request merged**